app/javascript/src/question_autocomplete.js

```diff
@@ -32,29 +31,6 @@ class AutocompleteQuestion extends Question {
     }, config));
 
     $node.addClass("AutocompleteQuestion");
-
-    const apiPath = this.menu.$items.filter('[data-action="upload"]').data('api-path');
-    const $uploadButton = $node.find('button[data-action="upload"]');
-   
-    $uploadButton.removeAttr('hidden');
-
-    $uploadButton.on('click', function() {
-      new DialogForm(apiPath, {
-        activator: $node.find('.fb-govuk-button-inverted'),
-        remote: true,
-        autoOpen: true,
-        submitOnClickSelector: 'input[type="submit"]',
-      onSuccess: function(data) {
-      },
```

Comment:
I'm getting a little confused over what is an autocomplete component (a question, it seems) and what is a DialogUploader (or the generic Dialog being used as one), so ignore my ignorance (and this comment) if I'm way off the mark, but this removal looks like the kind of thing that should be in a specific-use component, ie. why removed from here, and not in the controller (assuming it is related to the earlier comment about onSubmit/Success stuff in PagesController).

Reply:
OK, the reason all this is here, and then got removed is that there was originally another upload button on the page directly (as opposed to in the question menu)  hence me creating an autocomplete subclass and having this code.  

I guess the proper solution here is to tidy up fully and remove the autocomplete class.  Although we know there is a v2 of this work, and I can forsee the autocomplete question needing to have functionality other components don't so it might be worth leaving in.